validate.go

```diff
@@ -192,6 +194,48 @@ func checkProcess(process rspec.Process, rootfs string) {
 	}
 }
 
+func supportedMountTypes() (map[string]bool, error) {
+	supportedTypes := make(map[string]bool)
+	f, err := os.Open("/proc/filesystems")
```

Comment:
On 06/24/2016 01:41 AM, W. Trevor King wrote:
>> +	f, err := os.Open("/proc/filesystems")
> 
> This probably needs to go behind a |--host-checks| <https://github.com/opencontainers/ocitools/pull/49#issuecomment-216585596> (or |--host-specific| <https://github.com/opencontainers/ocitools/pull/49#issuecomment-220534023> or some such) flag, since reading the local system's |/proc/filesystems| is not portable <https://github.com/opencontainers/ocitools/pull/49#issuecomment-217793768>.
> 
Thanks for your suggestion.
I think using additional options like --host-checks or --host-specific is an option.

But currently supportedMountTypes() just supports Linux and I'm not sure does other systems has file like /proc/filesystems to
list system supported file systems? 
If user validated with --host-checks and used host with OS which is not Linux, the validation would still failed.
User may not want to get this result.

How about check host's OS first before executing checkMounts()?
If the host is Linux, then execute checkMounts(). If not, skip checkMounts().


-- 
Ma Shimiao
Development Dept.I
Nanjing Fujitsu Nanda Software Tech. Co., Ltd.(FNST)